File: sprint3/By languages/Java/lesson2code4.java
public static String asBinaryString(int n) {
    if (n < 0) {
        return "-" + asBinaryString(-n);
    }
    if (n == 0 || n == 1) {
        return Integer.toString(n);
    }
    int lastDigit = n % 2;
    return asBinaryString(n / 2) + Integer.toString(lastDigit);
}

File: sprint3/By languages/kotlin/lesson2code4.kt
fun as_binary_string(n: Int): String {
    if (n < 0) return "-" + as_binary_string(-n)
    if (n == 0 || n == 1) return n.toString()
    val lastDigit = n % 2
    return as_binary_string(n / 2) + lastDigit.toString()
}